code-challenges.test.js

```diff
@@ -15,24 +15,57 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+describe("shuffleItems", () => {
+    it("takes in an array, removes the first item from the array and shuffles the remaining items.", () => {
+        expect(shuffleItems(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+        expect(shuffleItems(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+})
+
+const colors1 = ["purple", "blue", "green", "yellow", "pink"]        
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
 const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
 // b) Create the function that makes the test pass.
-
+   
 // Pseudo code:
+// create a function that takes in the arrays
+const shuffleItems = (array) => {
+    // create a variable to remove purple from the array
+    let splitArray = array.splice(1)
+    // shuffle the array without purple using the fisher-yates method
+    for (let i = splitArray.length - 1; i > 0; i--){
+        let randomNum = Math.floor(Math.random() * (i + 1));
+        let whoKnows = splitArray[1]
+        // swap the elements in the split array
+        splitArray[1] = splitArray[randomNum]
+        splitArray[randomNum] = whoKnows
+    }
+    return splitArray
+}
+
+//if I am being completely honest. i have no clue what i just did. after the for loop, i had to just copy an example of the Fisher Yates Method to this function.
```

Comment:
Thanks for your honesty.  Fisher Yates is one way to solve this problem.  I like this explanation (as well as the inclusion of the .sort() method to randomize).  Hope this helps.  [fisher-yates](https://www.freecodecamp.org/news/how-to-shuffle-an-array-of-items-using-javascript-or-typescript/)